### DISCUSSION TITLE
Default values for out parameters

### DISCUSSION THREAD

**Discussion by ben-at-sparq** [dotnet/csharplang#7142] [Language Ideas] [answered]
I understand why a default value isn't allowed on a ref parameter, but it would seem an elegant solution to initialize out parameters.

Instead of this:

```
bool ParseSomething(out int x, out int y, out int z)
{
    x = 0;
    y = 0;
    z = 0;
    // do something
    return true;
}
```

Write this:

```
bool ParseSomething(out int x = 0, out int y = 0, out int z = 0)
{
    // do something
    return true;
}
```

I'm guessing this would be easy enough to add to lowering

**Comment by huoyaoyuan** [comment, marked as answer]
#1033 #3989

**Reply by ben-at-sparq** [reply]
Thanks! I did search but didn't find these - nice work
